homeassistant/components/blebox/binary_sensor.py

```diff
@@ -0,0 +1,62 @@
+"""BleBox sensor entities."""
+from dataclasses import dataclass
+
+from blebox_uniapi.binary_sensor import BinarySensor as BinarySensorFeature
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import BleBoxEntity, create_blebox_entities
+
+
+@dataclass
+class BleboxBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Class describing Blebox binary sensor entities."""
+
+
+BINARY_SENSOR_TYPES = (
+    BleboxBinarySensorEntityDescription(
+        key="moisture",
+        device_class=BinarySensorDeviceClass.MOISTURE,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a BleBox entry."""
+
+    create_blebox_entities(
+        hass,
+        config_entry,
+        async_add_entities,
+        BleBoxBinarySensorEntity,
+        "binary_sensors",
+    )
```

Comment:
I don't think you should delegate this to `__init__`

I think you should instead do something like this, so you can pass the description during initialisation:
```suggestion
    product: Box = hass.data[DOMAIN][config_entry.entry_id][PRODUCT]
    entities = []

    feature: Feature
    if features := product.features.get("binary_sensors"):
        for feature in features:
            for description in BINARY_SENSOR_TYPES:
                if description.key == feature.device_class:
                    entities.append(BleBoxBinarySensorEntity(feature, description))
                    break

    async_add_entities(entities, True)
```